ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -117,4 +119,40 @@ private static void throwOnUnknownProperty(
           + String.join(",", onlyInProvided));
     }
   }
+
+  public Optional<Character> getValueDelimiter() {
+    final String providedValueDelimiter = getString(CommonCreateConfigs.VALUE_DELIMITER_PROPERTY);
+    if (providedValueDelimiter == null) {
+      return Optional.empty();
+    }
+    if (providedValueDelimiter.isEmpty()) {
+      throw new KsqlException("Error in WITH clause property '"
+          + CommonCreateConfigs.VALUE_DELIMITER_PROPERTY
+          + "': Delimiter cannot be empty or whitespace."
+          + System.lineSeparator()
+          + "For tab or space delimited use 'TAB' or 'SPACE' as the delimeter."
+      );
+    } else if (providedValueDelimiter.length() == 1) {
+      return Optional.of(providedValueDelimiter.charAt(0));
+    } else {
+      final Character delim = NAMED_DELIMITERS.get(providedValueDelimiter);
+      if (delim != null) {
+        return Optional.of(delim);
+      } else {
```

Comment:
nit: unnecessary `else`.

Reply:
Imho, I think the else makes the intent of the code clearer.

Reply:
It leads to unnecessary indentation of the following block.  Given we have a code style on line width, I hope you can understand why we avoid such unnecessary `else` blocks.

Reply:
I prefer not to follow strict black and white rules such as "else's after returns are always bad" as whether they're appropriate or not really depends on context.
I do agree that in more complex situation with a bunch of tests and returns, else's could make the code harder to read and lead to line width issues as you point out.
However I don't think that's the case in this example and I think it makes it much easier to scan and for the reader to see what is happening.
Of course, if there is a strict project code convention regarding this I will follow it :)

Reply:
I'd rather leave this as is, as I think it's a bit religious what approach is better ;)